ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -240,6 +240,18 @@ StreamInfo remove(ClientInvocationId key) {
     return f;
   }
 
+  private void removeDataStream(DataStreamRequestByteBuf request) {
+    try {
+      final ClientInvocationId invocationId = ClientInvocationId.valueOf(request.getClientId(), request.getStreamId());
+      final StreamInfo s = streams.remove(invocationId);
+      if (s != null) {
+        s.getDivision().getDataStreamMap().remove(invocationId);
```

Comment:
Just found that we could avoid s.getDivision() to throw IOException; see https://issues.apache.org/jira/secure/attachment/13039740/596_review.patch